editors/code/package.json

```diff
@@ -307,12 +307,24 @@
                     "exclusiveMinimum": true,
                     "description": "Number of syntax trees rust-analyzer keeps in memory"
                 },
-                "rust-analyzer.displayInlayHints": {
-                    "type": "boolean",
-                    "default": true,
+                "rust-analyzer.inlayHintOpts.displayType": {
+                    "type": "string",
+                    "enum": [
+                        "off",
+                        "typeHints",
+                        "parameterHints",
+                        "full"
+                    ],
+                    "enumDescriptions": [
+                        "No type inlay hints",
+                        "Type inlays hints only",
+                        "Parameter inlays hints only",
+                        "All inlay hints types"
+                    ],
```

Comment:
```suggestion
                "rust-analyzer.inlayHints": {
                    "type": "object",
                    "properties": {
                        "types": {
                            "type": "boolean",
                            "default": true,
                            "description": "Whether to show inlay type hints"
                        },
                        "parameterNames": {
                            "type": "boolean",
                            "default": true,
                            "description": "Whether to show function parameter names inlay hints at the call site"
                        }
                    }
```
I believe just a struct with `boolean`s will play well. This should be more user-friendly, since people tend to misspel enum string values to much (and we don't validate this on client-side, but misspelling of config keys is handled by vscode). This also gives us flexibility to add new kinds of inlay hints without having to create enum value for each of their combination (i.e. `off`, `full`, `inlay1`, `inlay1_and_inlay2`, `inlay1_and_inlay3` ... etc. this gives us a combinatorial explision)
And maybe this makes sense to use such a struct on the server-side as well @SomeoneToIgnore, @slyngbaek ?

Reply:
I converted from using the `Vec<InlayKind>` to an object with props. Instead of using a single object type I made several nested props so that they show up in the GUI.

Reply:
@SomeoneToIgnore You're right. I tested it before pushing this time.